smartsim/_core/mli/infrastructure/storage/backbone_feature_store.py

```diff
@@ -352,68 +575,145 @@ def __init__(
         comm_channel: CommChannelBase,
         backbone: BackboneFeatureStore,
         filters: t.Optional[t.List[EventCategory]] = None,
-        batch_timeout: t.Optional[float] = None,
+        name: t.Optional[str] = None,
+        event_handler: t.Optional[t.Callable[[EventBase], None]] = None,
     ) -> None:
         """Initialize the EventConsumer instance.
 
         :param comm_channel: Communications channel to listen to for events
         :param backbone: The MLI backbone feature store
         :param filters: A list of event types to deliver. when empty, all
         events will be delivered
-        :param timeout: Maximum time to wait for messages to arrive; may be overridden
-        on individual calls to `receive`
+        :param name: A user-friendly name for logging. If not provided, an
+        auto-generated GUID will be used
         :raises ValueError: If batch_timeout <= 0
         """
-        if batch_timeout is not None and batch_timeout <= 0:
-            raise ValueError("batch_timeout must be a non-zero, positive value")
-
         self._comm_channel = comm_channel
```

Comment:
I think I saw some other classes have attribute doc strings for even private attributes.  Do we want that here?

Reply:
Always. Added. Also added the same validation for timeout and tests for both.